### PR TITLE
✨ Detect fast-check PBT library for fuzz section

### PR DESCRIPTION
**Pull request by dubzzz**
#### What kind of change does this PR introduce?

(Is it a bug fix, **feature**, docs update, something else?)

As suggested at https://github.com/ossf/scorecard/issues/2792#issuecomment-1562007596, we add support for the detection of fast-check as a possible fuzzing solution.

I also adapted the documentation related to fuzzing accordingly.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

The JavaScript/TypeScript property based testing library called [fast-check](https://fast-check.dev) was nor recognized as being a valid fuzzing system. This PR makes sure that it will be an acceptable fuzzing option.

#### What is the new behavior (if this is a feature change)?**

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

Suggested at https://github.com/ossf/scorecard/issues/2792#issuecomment-1562007596, but not fixing the issue itself.

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Consider fast-check as a valid fuzzing solution
```

### CODE REVIEW

File: checks/raw/fuzzing.go [hunk 1]
```diff
@@ -87,6 +89,30 @@ var languageFuzzSpecs = map[clients.LanguageName]languageFuzzConfig{
...
 				"and test that specific properties are satisfied."),
 	},
+	// Fuzz patterns for JavaScript and TypeScript based on property-based testing.
+	//
+	// Based on the import of one of these packages:
+	// * https://fast-check.dev/
+	//
+	// This is not an exhaustive list.
+	clients.JavaScript: {
+		filePattern: "*.js",
```
Comment (dubzzz) [inline checks/raw/fuzzing.go:99]: Is there a simple way to also check for extensions such as cjs, mjs, jsx,cjsx, mjsx which are some alternate extensions for JavaScript code? Same need for TypeScript. Testing js only covers must of the cases, but adding these extra ones might extend the coverage a little bit more.

**Comment by naveensrinivasan** [comment]
Thank you! Can you please do this for updating the docs https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#updating-docs?

**Comment by dubzzz** [comment]
Of course, I'll do

**Comment by dubzzz** [comment]
@naveensrinivasan Is my last commit doing what you requested?

**Review by spencerschrock — APPROVED** [review]

**Comment by dubzzz** [comment]
Sorry for the force push, I just resolved a conflict on README.md

**Review by naveensrinivasan — APPROVED** [review]

**Comment by dubzzz** [comment]
@naveensrinivasan It seems that coverage blocks the merge. Anything I can do? It seems the new part is covered, no?

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan It seems that coverage blocks the merge. Anything I can do? It seems the new part is covered, no?

This is the reason "This branch is out-of-date with the base branch"

Can please you rebase with the main and force-push it?

**Pull request merged**